otx/algorithms/common/adapters/mmcv/utils/automatic_bs.py

```diff
@@ -84,19 +86,27 @@ def train_func_single_iter(batch_size):
 
     default_bs = _get_batch_size(cfg)
 
-    available_bs = adapt_torch_model_bs(
+    bs_search_ago = BsSearchAlgo(
         train_func=train_func_single_iter,
-        current_bs=default_bs,
-        trainset_size=len(datasets[0]),
+        default_bs=default_bs,
+        max_bs=len(datasets[0]),
     )
+    if not_increase:
+        new_batch_size = bs_search_ago.auto_decrease_batch_size()
+    else:
+        new_batch_size = bs_search_ago.find_big_enough_batch_size()
```

Comment:
Can we make it bs_search_algo.find() do the all and return the proper new_batch_size of all cases?

Reply:
It's hard to do, because `auto_decrease_batch_size` only finds batch size lower than default value and `find_big_enough_batch_size` finds batch size which uses almost GPU memory.